src/FakeItEasy.Analyzer/NonVirtualSetupAnalyzer.cs

```diff
@@ -0,0 +1,108 @@
+﻿namespace FakeItEasy.Analyzer
+{
+    using System;
+    using System.Collections.Immutable;
+    using System.Linq;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CSharp;
+    using Microsoft.CodeAnalysis.CSharp.Syntax;
+    using Microsoft.CodeAnalysis.Diagnostics;
+
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    class NonVirtualSetupAnalyzer : DiagnosticAnalyzer
+    {
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(DiagnosticDefinitions.NonVirtualSetup);
+
+        private static readonly ImmutableDictionary<string, DiagnosticDescriptor> DiagnosticsMap = CreateDiagnosticsMap();
+
+        public override void Initialize(AnalysisContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            context.RegisterSyntaxNodeAction(AnalyzeMethod,
+                SyntaxKind.InvocationExpression);
+
+            context.RegisterSyntaxNodeAction(AnalyzeProperty,
+                SyntaxKind.SimpleMemberAccessExpression);
+        }
+
+        public static void AnalyzeMethod(SyntaxNodeAnalysisContext context)
+        {
+            //Method so exclude non-methods and interfaces
+
+            AnalyzeCall<InvocationExpressionSyntax>(context, x => NotMethod(x) || Interface(x));
+        }
+
+        public static void AnalyzeProperty(SyntaxNodeAnalysisContext context)
+        {
+            //Propery so exclude methods
+
+            AnalyzeCall<MemberAccessExpressionSyntax>(context, x => Method(x));

```

Comment:
```c#
namespace TheNamespace
{
    using FakeItEasy;

    public class TheClass
    {
        public void Test()
        {
            var foo = A.Fake<IFoo>();
            A.CallTo(() => foo.Bar);
        }
    }

    public interface IFoo
    {
        int Bar{get;}
    }
}
```

is detected as an invalid use:

`Test0.cs(11,28): warning FakeItEasy0002: Non virtual member 'Bar' cannot be intercepted.`